File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/TaxBracketInfo.js
/**
 * Contains tax rate information for a tax bracket.
 */

function TaxBracketInfo() {

    this.Rate = 0;
    this.MaxBase = 0;
    this.CountyOverrideTax = 0;
    this.StateOverrideTax = 0;
    this.StateOverrideOn = false;
    this.CountyOverrideOn = false;

    /**
     * Sets the tax rate. Value cannot be negative. For rated taxes, value must 
     * be from 0 to 1. Otherwise, value may be larger.
     * @param rate Number
     * @return Reference to this object for method chaining. 
     */
    this.rate = function (rate) {
        this.Rate = rate;
        return this;
    }

    /**
     * Tax rate. Value cannot be negative. For rated taxes, value must be from 0
     * to 1. Otherwise, value may be larger.
     * @return Number
     */
    this.getRate = function () {
        return this.Rate;
    }

    /* Reference to this object for method chaining.*
     * Sets the tax rate. Value cannot be negative. For rated taxes, value must 
     * be from 0 to 1. Otherwise, value may be larger.
     * @param rate Number 
     */
    this.setRate = function (rate) {
        this.Rate = rate;
    }

    /**
     * Sets the maximum base that this rate applies to.
     * @param maxBase Number
     * @return Reference to this object for method chaining. 
     */
    this.maxBase = function (maxBase) {
        this.MaxBase = maxBase;
        return this;
    }

    /**
     * Maximum base that this rate applies to.
     * @return Number
     */
    this.getMaxBase = function () {
        return this.MaxBase;
    }

    /* Reference to this object for method chaining.*
     * Sets the maximum base that this rate applies to.
     * @param maxBase Number 
     */
    this.setMaxBase = function (maxBase) {
        this.MaxBase = maxBase;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the county level.
     * @param countyOverrideTax Number
     * @return Reference to this object for method chaining. 
     */
    this.countyOverrideTax = function (countyOverrideTax) {
        this.CountyOverrideTax = countyOverrideTax;
        return this;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the county level.
     * @return Number
     */
    this.getCountyOverrideTax = function () {
        return this.CountyOverrideTax;
    }

    /* Reference to this object for method chaining.*
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the county level.
     * @param countyOverrideTax Number 
     */
    this.setCountyOverrideTax = function (countyOverrideTax) {
        this.CountyOverrideTax = countyOverrideTax;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the state level.
     * @param stateOverrideTax Number
     * @return Reference to this object for method chaining. 
     */
    this.stateOverrideTax = function (stateOverrideTax) {
        this.StateOverrideTax = stateOverrideTax;
        return this;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the state level.
     * @return Number
     */
    this.getStateOverrideTax = function () {
        return this.StateOverrideTax;
    }

    /* Reference to this object for method chaining.*
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the state level.
     * @param stateOverrideTax Number 
     */
    this.setStateOverrideTax = function (stateOverrideTax) {
        this.StateOverrideTax = stateOverrideTax;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), sets a value indicating 
     * that this tax replaces the tax at the state level.
     * @param stateOverrideOn Boolean
     * @return Reference to this object for method chaining. 
     */
    this.stateOverrideOn = function (stateOverrideOn) {
        this.StateOverrideOn = stateOverrideOn;
        return this;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49) indicates that this tax
     * replaces the tax at the state level.
     * @return Boolean
     */
    this.getStateOverrideOn = function () {
        return this.StateOverrideOn;
    }

    /* Reference to this object for method chaining.*
     * For sales and use taxes only (tax types 1 and 49), sets a value indicating 
     * that this tax replaces the tax at the state level.
     * @param stateOverrideOn Boolean 
     */
    this.setStateOverrideOn = function (stateOverrideOn) {
        this.StateOverrideOn = stateOverrideOn;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), sets a value indicating 
     * that this tax replaces the tax at the county level.
     * @param countyOverrideOn Boolean
     * @return Reference to this object for method chaining. 
     */
    this.countyOverrideOn = function (countyOverrideOn) {
        this.CountyOverrideOn = countyOverrideOn;
        return this;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), indicates that this
     * tax replaces the tax at the county level.
     * @return Boolean
     */
    this.getCountyOverrideOn = function () {
        return this.CountyOverrideOn;
    }

    /* Reference to this object for method chaining.*
     * For sales and use taxes only (tax types 1 and 49), sets a value indicating 
     * that this tax replaces the tax at the county level.
     * @param countyOverrideOn Boolean 
     */
    this.setCountyOverrideOn = function (countyOverrideOn) {
        this.CountyOverrideOn = countyOverrideOn;
    }
}

File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/TaxBracketInfo.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;

/**
 * Contains tax rate information for a tax bracket.
 */
public class TaxBracketInfo {

    @SerializedName("Rate")
    private Double rate = null;

    @SerializedName("MaxBase")
    private Double maxBase = null;

    @SerializedName("CountyOverrideTax")
    private Double countyOverrideTax = null;

    @SerializedName("StateOverrideTax")
    private Double stateOverrideTax = null;

    @SerializedName("StateOverrideOn")
    private Boolean stateOverrideOn = null;

    @SerializedName("CountyOverrideOn")
    private Boolean countyOverrideOn = null;

    /**
     * Sets the tax rate. Value cannot be negative. For rated taxes, value must 
     * be from 0 to 1. Otherwise, value may be larger.
     * @param rate
     * @return 
     */
    public TaxBracketInfo rate(Double rate) {
        this.rate = rate;
        return this;
    }

    /**
     * Tax rate. Value cannot be negative. For rated taxes, value must be from 0
     * to 1. Otherwise, value may be larger.
     * @return rate
     */
    public Double getRate() {
        return rate;
    }

    /**
     * Sets the tax rate. Value cannot be negative. For rated taxes, value must 
     * be from 0 to 1. Otherwise, value may be larger.
     * @param rate 
     */
    public void setRate(Double rate) {
        this.rate = rate;
    }

    /**
     * Sets the maximum base that this rate applies to.
     * @param maxBase
     * @return 
     */
    public TaxBracketInfo maxBase(Double maxBase) {
        this.maxBase = maxBase;
        return this;
    }

    /**
     * Maximum base that this rate applies to.
     * @return maxBase
     */
    public Double getMaxBase() {
        return maxBase;
    }

    /**
     * Sets the maximum base that this rate applies to.
     * @param maxBase 
     */
    public void setMaxBase(Double maxBase) {
        this.maxBase = maxBase;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the county level.
     * @param countyOverrideTax
     * @return 
     */
    public TaxBracketInfo countyOverrideTax(Double countyOverrideTax) {
        this.countyOverrideTax = countyOverrideTax;
        return this;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the county level.
     * @return countyOverrideTax
     */
    public Double getCountyOverrideTax() {
        return countyOverrideTax;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the county level.
     * @param countyOverrideTax 
     */
    public void setCountyOverrideTax(Double countyOverrideTax) {
        this.countyOverrideTax = countyOverrideTax;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the state level.
     * @param stateOverrideTax
     * @return 
     */
    public TaxBracketInfo stateOverrideTax(Double stateOverrideTax) {
        this.stateOverrideTax = stateOverrideTax;
        return this;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the state level.
     * @return stateOverrideTax
     */
    public Double getStateOverrideTax() {
        return stateOverrideTax;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), overrides the rate for
     * the tax at the state level.
     * @param stateOverrideTax 
     */
    public void setStateOverrideTax(Double stateOverrideTax) {
        this.stateOverrideTax = stateOverrideTax;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), sets a value indicating 
     * that this tax replaces the tax at the state level.
     * @param stateOverrideOn
     * @return 
     */
    public TaxBracketInfo stateOverrideOn(Boolean stateOverrideOn) {
        this.stateOverrideOn = stateOverrideOn;
        return this;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49) indicates that this tax
     * replaces the tax at the state level.
     * @return stateOverrideOn=
     */
    public Boolean getStateOverrideOn() {
        return stateOverrideOn;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), sets a value indicating 
     * that this tax replaces the tax at the state level.
     * @param stateOverrideOn 
     */
    public void setStateOverrideOn(Boolean stateOverrideOn) {
        this.stateOverrideOn = stateOverrideOn;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), sets a value indicating 
     * that this tax replaces the tax at the county level.
     * @param countyOverrideOn
     * @return 
     */
    public TaxBracketInfo countyOverrideOn(Boolean countyOverrideOn) {
        this.countyOverrideOn = countyOverrideOn;
        return this;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), indicates that this
     * tax replaces the tax at the county level.
     * @return countyOverrideOn
     */
    public Boolean getCountyOverrideOn() {
        return countyOverrideOn;
    }

    /**
     * For sales and use taxes only (tax types 1 and 49), sets a value indicating 
     * that this tax replaces the tax at the county level.
     * @param countyOverrideOn 
     */
    public void setCountyOverrideOn(Boolean countyOverrideOn) {
        this.countyOverrideOn = countyOverrideOn;
    }
}
